src/dev/MAX31855.cpp

```diff
@@ -7,21 +7,61 @@ namespace TMU::DEV {
 
 MAX31855::MAX31855(SPI& spi, uint8_t device) : spi(spi), device(device) {}
 
-uint16_t MAX31855::readTemp() {
-    uint16_t temp;
+MAX31855::MaxStatus MAX31855::readTemp(uint16_t& temp) {
+    uint16_t returned_data = 0;
 
     const uint8_t length = 4;
     uint8_t bytes[length];
 
     spi.startTransmission(device);
-    spi.read(bytes, length);
+    SPI::SPIStatus status = spi.read(bytes, length);
+    if (status != SPI::SPIStatus::OK) {
+        return MaxStatus::SPI_ERROR;
+    }
     spi.endTransmission(device);
 
-    temp = (((uint16_t) bytes[0]) << 8) | bytes[1];
+    returned_data = (((uint16_t) bytes[0]) << 8) | bytes[1];
```

Comment:
This declaration should probably be moved down to where the other `returned_data` calculations are happening. I think the variable should also be renamed to `returnedData` to match the camel case from the rest of the project.